zilliqa/src/consensus.rs

```diff
@@ -907,7 +907,7 @@ impl Consensus {
     }
 
     pub fn try_to_propose_new_block(&mut self) -> Result<Option<NetworkMessage>> {
-        if self.create_next_block_on_timeout {
+        if self.create_next_block_on_timeout && self.transaction_pool.has_txn_for_next_block() {
```

Comment:
Some extra comments here might be useful. I got a bit confused working out what this function does and why it needed the extra condition.